Makefile

```diff
@@ -439,8 +439,8 @@ requirements: virtualenv .sdist-requirements install-runners
 	@echo
 	# Make sure we use latest version of pip which is 19
 	$(VIRTUALENV_DIR)/bin/pip --version
-	$(VIRTUALENV_DIR)/bin/pip install --upgrade "pip==19.3.1"
-	$(VIRTUALENV_DIR)/bin/pip install --upgrade "setuptools==41.0.1"  # Required for packs.install in dev envs
+	$(VIRTUALENV_DIR)/bin/pip install --upgrade "pip>=19.3.1"
```

Comment:
Are both pip and setuptools version change required for this change?

We pinned both pip and setuptools in many places for the reasons based on experience of bad surprises.
If there is a motivation to upgrade it, - please pin to specific version and this should be changed in many repos across the org, incl. packaging and build Dockerfiles.

Reply:
Fixed.

Reply:
I’d be a fan up upgrading pip and setup tools right after release to work out bugs, I agree with @arma they are pinned for a reason, but also setup tools v20.0.x is way better.

Reply:
I agree. Let's create an issue in `stackstorm/st2` to track upgrading both pip + virtualenvs across the repos in next release and as another series of PRs.

Reply:
Pinning to pip version interval and avoiding relying on latest automatically was part of my review/feedback for this PR.
Moving this pip version change from `Makefile` to the `.travis.yml` is not a good way to address that feedback.

I'm wondering why this comment was marked as _Resolved_.

Reply:
I changed this back for you and marked it as resolved, then re-reverted my change and forgot to unresolve it. Sorry!

But, as per our Slack conversation, I consider pinning to exact pip and setuptools versions to be a code smell, and I'm against doing so until we have exhausted and documented every other possible workaround.

Reply:
As said before, if you're planning the change which finds an issue to reach the consensus, - do that as a proposal discussion to change the existing behavior. If TSC votes for the new change, - it's best to plan it for the next release and update consistently across all the repositories.

Before that happens, the building mechanisms should persist, especially before the release where any major changes or experiments should be discouraged and codebase freezed.

Reply:
Proposal discussion is in StackStorm/discussions#2.

I'll revert this back to what it was pending the conclusion of that discussion.